web/src/auth/LoginPage.js

```diff
@@ -376,26 +377,32 @@ class LoginPage extends React.Component {
       return Util.renderMessageLarge(this, this.state.msg);
     }
 
-    return (
-      <Row>
-        <Col span={24} style={{display: "flex", justifyContent: "center"}}>
-          <div style={{marginTop: "80px", marginBottom: "50px", textAlign: "center"}}>
-            {
-              Setting.renderHelmet(application)
-            }
-            {
-              Setting.renderLogo(application)
-            }
-            {/*{*/}
-            {/*  this.state.clientId !== null ? "Redirect" : null*/}
-            {/*}*/}
-            {
-              this.renderForm(application)
-            }
-          </div>
-        </Col>
-      </Row>
-    )
+      if ((this.state.application && this.state.application.loginPageCustom !== "")) {
+          return <LoginPageCustom Parent={this} Application={this.state.application} Html={this.state.application.loginPageCustom}/>
+      } else if (this.state.classes.application && this.state.classes.application.loginPageCustom !== "") {
+          return <LoginPageCustom Parent={this} Application={this.state.classes.application } Html={this.state.classes.application.loginPageCustom}/>
```

Comment:
`if (this.state.application?.loginPageCustom  !== "")`, other than `app&&app.loginPageCustom`